tests/config_generator.go

```diff
@@ -0,0 +1,144 @@
+package tests
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/flanksource/duty/models"
+	"github.com/google/uuid"
+	"github.com/samber/lo"
+	"gorm.io/gorm"
+)
+
+type ConfigGenerator struct {
+	PodPerDeployment, DeploymentPerNamespace, Namespaces, Nodes int
+	NumChangesPerConfig                                         int
+	NumInsightsPerConfig                                        int
+	HealthyPercentage                                           int
+	UnhealthyPercentage                                         int
+	UnknownPercentage                                           int
+	DeletedPercentage                                           int
+	Tags                                                        map[string]string
+	Generated                                                   struct {
+		Configs       []models.ConfigItem
+		Changes       []models.ConfigChange
+		Analysis      []models.ConfigAnalysis
+		Relationships []models.ConfigRelationship
+	}
+	count int
+}
+
+func (generator *ConfigGenerator) GenerateStatus() *string {
+	randomNumber := rand.Intn(100)
+
+	if randomNumber <= generator.UnhealthyPercentage {
+		return lo.ToPtr("unhealthy")
+	} else if randomNumber <= generator.UnhealthyPercentage+generator.UnknownPercentage {
+		return lo.ToPtr("unknown")
+	} else {
+		return lo.ToPtr("healthy")
+	}
+}
+
+func (generator *ConfigGenerator) GenerateDeleted() *time.Time {
+	randomNumber := rand.Intn(100)
+
+	if randomNumber <= generator.DeletedPercentage {
+		return lo.ToPtr(time.Now().Add(-10 * 24 * time.Hour))
+	}
+	return nil
+}
+
+func (generator *ConfigGenerator) GenerateConfigItem(configType string, parent *models.ConfigItem) models.ConfigItem {
+	changes := []models.ConfigChange{}
+	analysis := []models.ConfigAnalysis{}
+	generator.count++
+	name := fmt.Sprintf("%s-%d", strings.Split(configType, "::")[1], generator.count)
+
+	item := models.ConfigItem{
+		ID:        uuid.New(),
+		DeletedAt: generator.GenerateDeleted(),
```

Comment:
@moshloop I think we want this to be deterministic to be used in tests.

Reply:
Ok - So maybe setup a real structure so that the deployment have 1 active replicase with a pod, and 3 replicasets scaled to zero, each with deleted pods